src/Chapter2.hs

```diff
@@ -727,8 +755,13 @@ value of the element itself
 
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
+-- smartReplicate :: [Int] -> [Int]
+-- smartReplicate [] = []
+-- smartReplicate (x:xs) = (replicate x x) ++ smartReplicate xs
+-- dont understod how to use map smartReplicate arr = map (\x -> replicate x x)  arr
```

Comment:
I see that on the final solution below, you eventually used `concatMap` function that is basically both maps and concatenates (flattens) the resulting list. The problem with just `map` function is that it will return you the list of the lists, but what you need here is the flattened list, therefore usage of the `concat` function is needed 🙂 👍🏼

Reply:
I understood. Just wanted to make sure it was right to use ConcatMap. Thx